Notes.md

```diff
@@ -67,5 +76,35 @@ subtyping between types.
 For instance, `Int` could be a subtype of `Number` such that a function
 could declare to work with just `Number`s.
 
-[^swift-main]: Only the `main.swift` file can contain top-level code
-[^swift-variance]: Swift comes to mind here
+
+### Footnotes
+
+[^swift-main]: Only the `main.swift` file can contain top-level code   
+[^swift-variance]: Swift comes to mind here  
+[^scala-abstract-types-docs]: See [Scala's abstract types][AbstractTypes-Scala]  
+[^ocaml-module-system]: See [OCaml's module system][AbstractTypes-OCaml]  
+[^scala-abstract-types-explained]: See [this StackOverflow question][AbstractTypes-StackOverflow]
+which explains the effective difference between the parameterized generics and
+abstract types in Scala.
+
+Example from question:  
+>
+> ```
```

Comment:
I really like code to simply be indented by four spaces, but other than that I'll merge it